pkg/cli/admin/release/extract_tools.go

```diff
@@ -908,3 +927,90 @@ func copyAndReplace(errorOutput io.Writer, w io.Writer, r io.Reader, bufferSize
 		offset = end - writeTo
 	}
 }
+
+func findClusterIncludeConfig(ctx context.Context, installConfigPath string, restConfig *rest.Config) (requiredFeatureSet *string, profile *string, platform *string, capabilities *configv1.ClusterVersionCapabilitiesStatus, overrides []configv1.ComponentOverride, err error) {
+	clientVersion, reportedVersion, err := version.ExtractVersion()
+	if err != nil {
+		return nil, nil, nil, nil, nil, err
+	}
+	if reportedVersion == "" {
+		reportedVersion = clientVersion.String()
+	}
+
+	if installConfigPath != "" {
+		installConfigBytes, err := os.ReadFile(installConfigPath)
+		if err != nil {
+			return nil, nil, nil, nil, nil, err
+		}
+
+		data := installConfig{}
+
+		if err := yaml.Unmarshal(installConfigBytes, &data); err != nil {
+			return nil, nil, nil, nil, nil, fmt.Errorf("failed to parse %s: %w", installConfigPath, err)
+		}
+
+		if data.APIVersion != "v1" {
+			return nil, nil, nil, nil, nil, fmt.Errorf("unrecognized %s API version: %q (expected %q)", installConfigPath, data.APIVersion, "v1")
+		}
+
+		requiredFeatureSet = pointer.String(string(data.FeatureSet))
+		profile = pointer.String(manifest.DefaultClusterProfile) // assumption, but there's no install-config data about profile to give us more insight
+		for key := range data.Platform {
+			platform = pointer.String(key)
+		}
+
+		if data.Capabilities != nil {
+			capabilities = &configv1.ClusterVersionCapabilitiesStatus{}
+			if enabled, ok := configv1.ClusterVersionCapabilitySets[data.Capabilities.BaselineCapabilitySet]; !ok {
+				return nil, nil, nil, nil, nil, fmt.Errorf("unrecognized baselineCapabilitySet %q", data.Capabilities.BaselineCapabilitySet)
+			} else {
+				if data.Capabilities.BaselineCapabilitySet == configv1.ClusterVersionCapabilitySetCurrent {
+					klog.Infof("If the eventual cluster will not be the same minor version as this %s 'oc', the actual %s capability set may differ.", reportedVersion, data.Capabilities.BaselineCapabilitySet)
+				}
+				capabilities.EnabledCapabilities = append(capabilities.EnabledCapabilities, enabled...)
+			}
+			capabilities.EnabledCapabilities = append(capabilities.EnabledCapabilities, data.Capabilities.AdditionalEnabledCapabilities...)
+
+			klog.Infof("If the eventual cluster will not be the same minor version as this %s 'oc', the known capability sets may differ.", reportedVersion)
+			capabilities.KnownCapabilities = configv1.KnownClusterVersionCapabilities
+		}
+
+		return requiredFeatureSet, profile, platform, capabilities, overrides, nil
```

Comment:
I think we can extract all this to a `func includesFromInstallConfig(installConfigPath string) (*string, *string, *string, *configv1.ClusterVersionCapabilitiesStatus, []configv1.ComponentOverride, error)` method to reduce complexity... And with an even simpler signature if we end up collecting these bits in to a struct! :)

Reply:
Refactored into an install-config function and a REST-config function with e82184c92 -> ee48493c0.